Apps/W1/Shopify/app/src/Base/Tables/ShpfyShop.Table.al

```diff
@@ -507,33 +553,46 @@ table 30102 "Shpfy Shop"
         end;
     end;
 
-    internal procedure GetLastSyncTime(Type: Enum "Shpfy Synchronization Type"): DateTime
+    internal procedure GetLastSyncTime(Type: enum "Shpfy Synchronization Type"): DateTime
     var
-        SyncInfo: Record "Shpfy Synchronization Info";
+        ShpfySynchronizationInfo: Record "Shpfy Synchronization Info";
     begin
-        if SyncInfo.Get(Rec.Code, Type) then
-            exit(SyncInfo."Last Sync Time");
+        if Type = "Shpfy Synchronization Type"::Orders then begin
+            if Rec."Shop Id" = 0 then begin
+                CalcShopId();
```

Comment:
now that we have these implicit with warnings some of the code becomes very confusing. 
here we have:
CalcShopId()
Rec.Modify()

now, potentially this would act on two different instances... Well, at least it's unclear what CalcShopId() could possibly do to the Rec that it deserves a modify.

Then you look at that method and there's not a rec in sight which adds to the confusion.

So:
first, Rec. should be added everywhere. simple.

Should the code be:
Rec.CalcShopId();
Rec.Modify();

well, yes... 

and yes i know it makes no difference to the execution but it is much easier on the eyes...